sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -235,8 +236,17 @@ class ResolveSessionCatalog(
     case DescribeRelation(ResolvedV1TableOrViewIdentifier(ident), partitionSpec, isExtended) =>
       DescribeTableCommand(ident.asTableIdentifier, partitionSpec, isExtended)
 
-    case DescribeColumn(ResolvedV1TableOrViewIdentifier(ident), colNameParts, isExtended) =>
-      DescribeColumnCommand(ident.asTableIdentifier, colNameParts, isExtended)
+    case DescribeColumn(ResolvedV1TableOrViewIdentifier(ident), column, isExtended) =>
+      column match {
+        case u: UnresolvedAttribute =>
+          // For views, the column will not be resolved by `ResolveReferences` because
+          // `ResolvedView` stores only the identifier.
+          DescribeColumnCommand(ident.asTableIdentifier, u.nameParts, isExtended)
```

Comment:
Is it possible there is unresolved attribute but the `relation` of `DescribeColumn` is a v1 table?

Reply:
It's possible when the column name doesn't exist in the table, and we should give a clear error message: `Column $colName does not exist`

Reply:
The comment was confusing since `DescribeColumnCommand` resolves the column again. I updated to separate view and table matching to make the intention clear. Thanks.